src/retryOnEmpty.ts

```diff
@@ -103,41 +98,39 @@ export function createRetryOnEmptyMiddleware({
     // create child request with specific block-ref
     const childRequest = klona(req);
     // attempt child request until non-empty response is received
-    const childResponse: PendingJsonRpcResponse<Block> = await retry(
-      10,
-      async () => {
+    try {
+      const childResponse = await retry(10, async () => {
         log('Performing request %o', childRequest);
-        const attemptResponse: PendingJsonRpcResponse<Block> = await pify(
-          provider.sendAsync,
-        ).call(provider, childRequest);
-        log('Response is %o', attemptResponse);
+        const result = await provider.request<JsonRpcParams, Block>(
+          childRequest,
+        );
+        log('Result is %o', result);
         // verify result
-        if (emptyValues.includes(attemptResponse.result as any)) {
+        if (emptyValues.includes(result as any)) {
           throw new Error(
-            `RetryOnEmptyMiddleware - empty response "${JSON.stringify(
-              attemptResponse,
+            `RetryOnEmptyMiddleware - empty result "${JSON.stringify(
+              result,
             )}" for request "${JSON.stringify(childRequest)}"`,
           );
         }
-        return attemptResponse;
-      },
-    );
-    log(
-      'Copying result %o and error %o',
-      childResponse.result,
-      childResponse.error,
-    );
-    // copy child response onto original response
-    res.result = childResponse.result;
-    res.error = childResponse.error;
+        return result;
+      });
+      log('Copying result %o', childResponse.result);
+      res.result = childResponse;
+      res.error = undefined;
+    } catch (error: any) {
```

Comment:
Similar comment as above: do we need the type annotation?

``` suggestion
    } catch (error) {
```

Or, alternatively, can we use `unknown`?


``` suggestion
    } catch (error: unknown) {
```